synapse/logging/opentracing.py

```diff
@@ -910,10 +910,10 @@ def _wrapping_logic(func: Callable[P, R], *args: P.args, **kwargs: P.kwargs) ->
         async def _wrapper(
             *args: P.args, **kwargs: P.kwargs
         ) -> Any:  # Return type is RInner
-            with wrapping_logic(func, *args, **kwargs):
-                # type-ignore: func() returns R, but mypy doesn't know that R is
-                # Awaitable here.
-                return await func(*args, **kwargs)  # type: ignore[misc]
+            # type-ignore: func() returns R, but mypy doesn't know that R is
+            # Awaitable here.
+            with wrapping_logic(func, *args, **kwargs):  # type: ignore[arg-type]
```

Comment:
Moving this comment might not have made perfect sense, but the tl;dr is that mypy ends up with:

```
synapse/logging/opentracing.py:916: error: Argument 1 has incompatible type "Callable[P, CoroutineType[Any, Any, Any]]"; expected "Callable[P, R]"  [arg-type]
```

So it doesn't know that `func(...)` will return `Awaitable[R]`.